lib/pages/child_menu/customizable_column.dart

```diff
@@ -113,47 +124,121 @@ class _CustomizableColumnState extends State<CustomizableColumn> {
-      appBar: AppBar(
-        title: Text("Child Mode"),
-        automaticallyImplyLeading: false,
-        actions: [
-          Padding(
-            padding: const EdgeInsets.only(right: 20.0),
-            child: GestureDetector(
-                key: Key("logoutButton"),
-                //only triggers when its pressed for some time and swiped up
-                onLongPressUp: () async {
-                  openLogoutDialog(context);
+    if (!widget.mock) {
+      FirebaseFunctions firebaseFunctions = FirebaseFunctions(
+          auth: FirebaseAuth.instance,
+          firestore: FirebaseFirestore.instance,
+          storage: FirebaseStorage.instance);
+      return Scaffold(
+        appBar: AppBar(
+          title: Text("Child Mode"),
+          automaticallyImplyLeading: false,
+          actions: [
+            Padding(
+              padding: const EdgeInsets.only(right: 20.0),
+              child: GestureDetector(
+                  key: Key("logoutButton"),
+                  //only triggers when its pressed for some time and swiped up
+                  onLongPressUp: () async {
+                    openLogoutDialog(context);
+                  },
+                  child: Icon(Icons.exit_to_app)),
+            )
+          ],
+        ),
+        body: FutureBuilder(
+          future:
+              getListFromChoiceBoards(), //retrieves a list from the database of categories and items associated with the category
+          builder: (context, snapshot) {
+            if (snapshot.hasData) {
+              List<List<ClickableImage>> categories =
+                  snapshot.data as List<List<ClickableImage>>;
+              List<List<ClickableImage>> filtered = filterImages(categories);
+              return ListView.separated(
+                itemBuilder: (context, index) {
+                  return CustomizableRow(
+                    key: Key("row$index"),
+                    categoryTitle: filtered[index][0].name,
+                    imagePreviews: filtered[index],
+                    unfilteredImages: categories[index],
+                  );
+                },
+                itemCount: categories.length,
+                separatorBuilder: (context, index) {
+                  return Divider(height: 2);
                 },
-                child: Icon(Icons.exit_to_app)),
-          )
-        ],
-      ),
-      body: ListView.separated(
-        itemBuilder: (context, index) {
-          return CustomizableRow(
-            key: Key("row$index"),
-            categoryTitle: rowConfigs[index]['categoryTitle'],
-            imagePreviews: rowConfigs[index]['images'],
-          );
-        },
-        itemCount: rowConfigs.length,
-        separatorBuilder: (context, index) {
-          return Divider(height: 2);
-        },
-      ),
-    );
+              );
+            } else {
+              return Center(child: CircularProgressIndicator());
+            }
+          },
+        ),
+      );
+    } else {
```

Comment:
just realized how large this if statement is. Please do not do that, if the view you are testing is 100% different from the displayed view then there isn't much point to testing it. Please refactor the code to use essentially the same view for both testing and normal purposes. The differences should happen either by mocking the connection to the database or by not doing the database calls and passing in a fake list. But the build() method can not be that different for testing and normal situations, it defeats the purpose